src/node.rs

```diff
@@ -19,7 +19,7 @@ impl<'a> Node<'a> {
         self.allocator.sexp(self.node)
     }
 
-    pub fn atom(&'a self) -> Option<&'a [u8]> {
```

Comment:
this was a mistake. The returned buffer depends on the allocator in `self` (which is what `'a` refers to in the outer scope). This introduces a new `'a`, referring to `self`'s lifetime, shadowing the correct one.